### PR TITLE
[BE][SSO] Migrate to Knex ``DELETE dashboard/users`` endpoints

### PR DESCRIPTION
**Pull request by Aredhel269**

### CODE REVIEW

**Review by ghOdisea — DISMISSED** [review]

File: services/sso/src/controllers/dashboard/users/batchDelete.ts [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
+  if (!Array.isArray(ids) || ids.length === 0) {
     throw new NotFoundError('No user found')
   }
-  const verifyQuery = `
-  SELECT id, deleted_at FROM "user" WHERE id = ANY($1::text[]);
-  `
-  const verifyResult = await client.query(verifyQuery, [ids])
-  const existingIds = verifyResult.rows.map((row) => row.id)
+
+  // Verificar quins usuaris existeixen a la base de dades
```
Comment (kevinmamaqi) [inline services/sso/src/controllers/dashboard/users/batchDelete.ts:11]: Eliminar

File: services/sso/src/controllers/dashboard/users/batchDelete.ts [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
     throw new NotFoundError('No user found')
   }
 
-  const toBeDeletedUser: string[] = []
-  verifyResult.rows.forEach((row) => {
-    if (row.deleted_at === null) {
-      toBeDeletedUser.push(row.id)
-    }
-  })
+  // Filtrar els usuaris que no han estat esborrats
```
Comment (kevinmamaqi) [inline services/sso/src/controllers/dashboard/users/batchDelete.ts:22]: Eliminar

File: services/sso/src/controllers/dashboard/users/batchDelete.ts [hunk 3]
```diff
@@ -1,35 +1,36 @@
...
+  const toBeDeletedUser: string[] = verifyResult
+    .filter((row) => row.deleted_at === null)
+    .map((row) => row.id)
 
+  // Si hi ha usuaris per esborrar, actualitzar la seva columna 'deleted_at'
```
Comment (kevinmamaqi) [inline services/sso/src/controllers/dashboard/users/batchDelete.ts:27]: Eliminar

File: services/sso/src/controllers/dashboard/users/batchDelete.ts [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
-      SET deleted_at = CURRENT_TIMESTAMP
-      WHERE id = ANY($1::text[])
-    `
-    await client.query(deleteQuery, [toBeDeletedUser])
+    await db('user')
+      .whereIn('id', toBeDeletedUser)
+      .update({ deleted_at: db.fn.now() })
   }
+
+  // Resposta 204: No Content, perquè no hi ha cos a la resposta
```
Comment (kevinmamaqi) [inline services/sso/src/controllers/dashboard/users/batchDelete.ts:34]: Eliminar

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

**Review by kevinmamaqi — APPROVED** [review]

**Review by ghOdisea — APPROVED** [review]

**Pull request merged**